Source/Mosa.DeviceDriver/ISA/SoundBlaster16.cs

```diff
@@ -0,0 +1,78 @@
+
+		public override void Initialize()
+		{
+			MixerPort = Device.Resources.GetIOPortWrite(0, 0);
+			MixerDataPort = Device.Resources.GetIOPortWrite(0, 1);
+			Reset = Device.Resources.GetIOPortWrite(0, 2);
+			Read = Device.Resources.GetIOPortRead(0, 6);
+			Write = Device.Resources.GetIOPortWrite(0, 8);
+
+			ChannelControl = HAL.GetWriteIOPort(Play16BitSongs ? (ushort)0xD4 : (ushort)0x0A);
+			FlipFlop = HAL.GetWriteIOPort(Play16BitSongs ? (ushort)0xD8 : (ushort)0x0C);
+			TransferMode = HAL.GetWriteIOPort(Play16BitSongs ? (ushort)0xD6 : (ushort)0x0B);
+			PageNumber = HAL.GetWriteIOPort(Play16BitSongs ? (ushort)0x8B : (ushort)0x83);
+			Position = HAL.GetWriteIOPort(Play16BitSongs ? (ushort)0xC4 : (ushort)0x02);
+			DataLength = HAL.GetWriteIOPort(Play16BitSongs ? (ushort)0xC6 : (ushort)0x03);
+
+			Reset.Write8(1);
+			HAL.Pause();
+			Reset.Write8(0);
+
+			_ = Read.Read8();
+
+            // Turn the speaker on
+            Write.Write8((byte)Options.TurnOn);
+            SetVolume(0xF); // Max volume
+        }
+
+        public void SetVolume(byte v)
```

Comment:
Change to IAudioDevice.SetVolume

Reply:
I can't, because SetVolume is used just above (SetVolume(0xF)), and if I change to what you said it doesn't recognize the function anymore (it says it doesn't find it)

Reply:
It'll find it, if casted to the interface type.

Reply:
So like this? ``(this as IAudioDevice).SetVolume(0xF)``